pkg/logs/input/container/scanner.go

```diff
@@ -83,16 +111,14 @@ func (s *Scanner) scan(tailFromBeginning bool) {
 	for _, container := range runningContainers {
 		for _, source := range s.sources {
 			if s.sourceShouldMonitorContainer(source, container) {
-				containersToMonitor[container.ID] = true
-
 				tailer, isTailed := s.tailers[container.ID]
 				if isTailed && tailer.shouldStop {
-					s.stopTailer(tailer)
-					isTailed = false
+					continue
```

Comment:
why are we changing this?

Reply:
All tailers that are not in `containersToMonitor ` are stopped afterward. There is no need to keep a tailer in `containersToMonitor ` and to stop it afterward, WDYT ?

Reply:
The issue here could be that if a tailer needs to be restarted, it would get dropped at first iteration, and then get started at second iteration 10 seconds later. However as containers don't logrotate, I don't think this is a real problem, so we can disregard this comment